src/preprocess.rkt

```diff
@@ -29,8 +37,12 @@
      (list*
       initial
       specification
-      (append even-identities swap-identities))
+      (append even-identities odd-identities swap-identities))
      (*simplify-rules*)))
+  (define (split-others others)
+    (define-values (evens others*) (split-at others (length even-identities)))
+    (define-values (odds swaps) (split-at others* (length odd-identities)))
+    (values evens odds swaps))
```

Comment:
Clearly long-term we'll want something else here (as in, this code won't generalize to even more preprocessing) but I think it's fine for now.